source/Octostache/Templates/Functions/HashFunction.cs

```diff
@@ -0,0 +1,156 @@
+
+            try
+            {
+                var argumentBytes = hashOptions.GetBytes(argument);
+
+                var hsh = algorithm();
+                return HexDigest(hsh.ComputeHash(argumentBytes), hashOptions);
+            }
+            catch
+            {
+                // Likely invalid input
+                return null;
+            }
+        }
+
+        static string HexDigest(byte[] bytes, HashOptions options)
+        {
+            var size = options.DigestSize.GetValueOrDefault(bytes.Length);
+            if (size > bytes.Length)
+            {
+                size = bytes.Length;
+            }
+            var sb = new StringBuilder(size * 2);
+
+            for (var i = 0; i < size; i++)
+            {
+                sb.Append(bytes[i].ToString("x2"));
+            }
+
+            return sb.ToString();
+        }
+
+        class HashOptions
+        {
+            public HashOptions(string[] options)
+            {
+                if (options.Length == 0)
+                {
+                    return;
+                }
+                if (options.Length > 2)
+                {
+                    IsValid = false;
+                    return;
+                }
+
+                if (int.TryParse(options[0], out var size) && size > 0)
+                {
+                    DigestSize = size;
+                    if (options.Length > 1)
+                    {
+                        var encoding = GetEncoding(options[1]);
+                        if (encoding == null)
+                        {
+                            IsValid = false;
+                        }
+                        else
+                        {
+                            GetBytes = encoding;
+                        }
+                    }
+                }
+                else
+                {
+                    var encoding = GetEncoding(options[0]);
+                    if (encoding == null)
+                    {
+                        IsValid = false;
+                    }
+                    else
+                    {
+                        GetBytes = encoding;
+                    }
+
+                    if (IsValid && options.Length > 1)
+                    {
+                        if (int.TryParse(options[1], out size) && size > 0)
+                        {
+                            DigestSize = size;
+                        }
+                        else
+                        {
+                            IsValid = false;
+                        }
+                    }
+                }
+            }
+
+            public Func<string, byte[]> GetBytes { get; } = Encoding.UTF8.GetBytes;
+            public int? DigestSize { get; }
+
+            public bool IsValid { get; } = true;
+
+            static Func<string, byte[]>? GetEncoding(string encoding)
+            {
+                switch (encoding.ToLowerInvariant())
+                {
+                    case "base64":
```

Comment:
Is it possible to use the existing `FromBase64` filter to achieve this?

Reply:
The `FromBase64` filter takes a base64 string and treats the underlying bytes as either Unicode or utf8 to get a string. This usage gets the raw bytes as that is what is used to calculate the hash. This way you could have a variable be binary content encoded in base64 but still be able to get the hash of the actual bytes (instead of the hash of the base64 representation of the bytes)